api/types/client.go

```diff
@@ -319,4 +321,54 @@ type VanClientInterface interface {
 	GetIngressDefault() string
 	RevokeAccess(ctx context.Context) error
 	NetworkStatus() ([]*SiteInfo, error)
+	ServiceManager(namespace string) Services
+	ConfigMapManager(namespace string) ConfigMaps
+	DeploymentManager(namespace string) Deployments
+	SecretManager(namespace string) Secrets
+	StatefulSetManager(namespace string) StatefulSets
+}
+
+type Services interface {
+	GetService(name string, options *v1.GetOptions) (*corev1.Service, bool, error)
+	DeleteService(svc *corev1.Service, options *v1.DeleteOptions) error
```

Comment:
Is there a specific need to expose DeleteOptions, GetOptions? Likewise for ListOptions, though there we clearly may need to request different types of list, is that the right level of abstraction?

Reply:
@grs I got rid of GetOptions and DeleteOptions already. Now I am going to improve on the ListOptions.